modules/message-lane/src/benchmarking.rs

```diff
@@ -200,6 +218,111 @@ benchmarks_instance! {
 		);
 	}
 
+	// Benchmark `receive_messages_delivery_proof` extrinsic with fllowing conditions:
+	// * single relayer is rewarded for relaying single message;
+	// * relayer account does not exist (in practice it needs to exist in production environment).
+	//
+	// This is base benchmark for all other confirmations delivery benchmarks.
+	receive_delivery_proof_for_single_message {
+		let relayers_fund_id = relayer_fund_account_id::<T, I>();
+		let relayer_id: T::AccountId = account("relayer", 0, SEED);
+		let relayer_balance = T::account_balance(&relayer_id);
+		T::endow_account(&relayers_fund_id);
+
+		// send message that we're going to confirm
+		send_regular_message::<T, I>();
+
+		let proof = T::prepare_message_delivery_proof(MessageDeliveryProofParams {
+			lane: bench_lane_id(),
+			inbound_lane_data: InboundLaneData {
+				relayers: vec![(1, 1, relayer_id.clone())].into_iter().collect(),
+				latest_received_nonce: 1,
+				latest_confirmed_nonce: 0,
+			}
+		});
+	}: receive_messages_delivery_proof(RawOrigin::Signed(relayer_id.clone()), proof)
+	verify {
+		assert_eq!(
+			T::account_balance(&relayer_id),
+			relayer_balance + MESSAGE_FEE.into(),
```

Comment:
So here we don't need to actually assert that `latest_confirmed_nonce = 1` since the fact that we got paid a fee is "proof" that it was confirmed?

Reply:
Yes, I don't think everything should be tested here - there are regular tests for that. But probably worth replacing this test with the test you've mentioned, because runtime may not pay rewards instantly (although now all runtimes are supposed to do that). Otoh probably switching from instant rewards to some other schema may require changes to weight formula, so better to fail test in this case...